journalpump/journalpump.py

```diff
@@ -377,10 +377,20 @@ def inc_line_stats(self, *, journal_lines, journal_bytes):
             sender.refresh_stats()
 
     def read_next(self):
+        """Read next message from journal reader."""
         if not self.journald_reader:
             return None
 
-        jobject: JournalObject = cast(JournalObject, next(self.journald_reader))
+        try:
+            jobject: JournalObject = cast(JournalObject, next(self.journald_reader))
+        except OSError as ex:
```

Comment:
How stable is the error message? `OSerror` is a very broad exception, it would make sense to check the exception message so that we handle only the ones related to the problem.